### ISSUE TITLE
Body parameter with x-examples causes JsonParseException

### ISSUE THREAD

**Issue by kevinoid** [swagger-api/swagger-codegen#4442]
##### Description

Running swagger-codegen on a Swagger definition which includes the `x-examples` property in a Parameter with `in: body` causes code generation to fail with `com.fasterxml.jackson.core.JsonParseException`.

##### Swagger-codegen version

master (ba285759)
The issue is also present in 2.2.1 and 2.1.6.

##### Swagger declaration file content

```yaml
swagger: "2.0"
info: 
  version: "1.0.0"
  title: API which causes JsonParseException
paths: 
  /pets: 
    post: 
      parameters: 
      - name: "pet"
        in: "body"
        schema:
          type: object
        x-examples:
          application/json:
            foo: bar
      responses:
        "200":
          description: Success
          schema:
            type: object
```

##### Steps to reproduce

With the above Swagger declaration file saved as `bad_example.yaml` run:
```sh
git clone https://github.com/swagger-api/swagger-codegen.git
cd swagger-codegen
mvn clean package
java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i bad_example.yaml -l java -o /tmp/result
```
Which produces the following output:

    [main] INFO io.swagger.parser.Swagger20Parser - reading from bad_example.yaml
    [main] INFO io.swagger.parser.Swagger20Parser - reading from bad_example.yaml
    com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'swagger': was expecting ('true', 'false' or 'null')
     at [Source: bad_example.yaml; line: 1, column: 9]
            at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1419)
            at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:508)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3201)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2360)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:794)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:690)
            at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3105)
            at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3051)
            at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:1861)
            at io.swagger.parser.SwaggerCompatConverter.readResourceListing(SwaggerCompatConverter.java:147)
            at io.swagger.parser.SwaggerCompatConverter.read(SwaggerCompatConverter.java:74)
            at io.swagger.parser.SwaggerParser.read(SwaggerParser.java:72)
            at io.swagger.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:391)
            at io.swagger.codegen.cmd.Generate.run(Generate.java:221)
            at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:41)
    Exception in thread "main" java.lang.RuntimeException: missing swagger input or config!
            at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:635)
            at io.swagger.codegen.cmd.Generate.run(Generate.java:223)
            at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:41)

(Note that the issue appears regardless of the chosen language.)

##### Related issues

The `JsonParseException: Unrecognized token 'swagger'` exception has occurred before due to bad encoding file encoding (#1575) or use of YAML anchors (swagger-api/swagger-parser#146).  The cause in both of those cases appears to be a Jackson issue.  Since the issue here only occurs when `x-examples` appears on specific Parameter types, that suggests to me that the issue is not solely with Jackson.

There is an open issue for `x-examples` support in Swagger UI (swagger-api/swagger-ui#1980), which is unrelated to this issue.  Whether or not that is implemented, `x-examples` should not prevent code generation.

**Comment by fehguy** [comment]
can you please open this in the swagger-parser project?

**Comment by kevinoid** [comment]
@fehguy sure thing.  Reported as swagger-api/swagger-parser#360.

**Comment by kevinoid** [comment]
This is fixed for me by swagger-api/swagger-parser#381 (after rebuilding with `swagger-parser-version` changed to `1.0.26-SNAPSHOT` to use the fix).  Thanks again!
